mlflow/gateway/app.py

```diff
@@ -53,81 +55,59 @@ async def search_routes():
     return {"routes": ACTIVE_ROUTES}
 
 
-async def _chat(request: chat.RequestPayload) -> chat.ResponsePayload:
-    return chat.ResponsePayload(
-        **{
-            "candidates": [
-                {
-                    "message": msg,
-                    "metadata": {},
-                }
-                for msg in request.messages
-            ],
-            "metadata": {
-                "input_tokens": 1,
-                "output_tokens": 2,
-                "total_tokens": 3,
-                "model": "gpt-3.5-turbo",
-                "route_type": "llm/v1/chat",
-            },
-        }
-    )
+def _create_chat_endpoint(config: RouteConfig):
+    prov = get_provider(config.model.provider)(config)
 
+    async def _chat(payload: chat.RequestPayload) -> chat.ResponsePayload:
+        return await prov.chat(payload)
 
-async def _completions(request: completions.RequestPayload) -> completions.ResponsePayload:
-    return completions.ResponsePayload(
-        **{
-            "candidates": [
-                {
-                    "text": request.prompt,
-                    "metadata": {},
-                },
-            ],
-            "metadata": {
-                "input_tokens": 1,
-                "output_tokens": 2,
-                "total_tokens": 3,
-                "model": "gpt-3.5-turbo",
-                "route_type": "llm/v1/completions",
-            },
-        }
-    )
+    return _chat
 
 
-async def _embeddings(_request: embeddings.RequestPayload) -> embeddings.ResponsePayload:
-    return embeddings.ResponsePayload(
-        **{
-            "embeddings": [0.1, 0.2, 0.3],
-            "metadata": {
-                "input_tokens": 1,
-                "output_tokens": 2,
-                "total_tokens": 3,
-                "model": "gpt-3.5-turbo",
-                "route_type": "llm/v1/embeddings",
-            },
-        }
-    )
+def _create_completions_endpoint(config: RouteConfig):
+    prov = get_provider(config.model.provider)(config)
+
```

Comment:
We can validate the provider supports `chat` here by using `prov.SUPPORTED_TASKS` if necessary.